build_tools/packaging/promote_from_rc_to_final.py

```diff
@@ -0,0 +1,278 @@
+        return False
+
+    print("  Starting to execute version change")
+    new_wheel_path = change_wheel_version.change_wheel_version(
+        pathlib.Path(filename),
+        str(base_version),
+        local_version,
+        callback_func=wheel_change_extra_files,
+    )
+    print("  Version change done")
+
+    new_wheel = Wheel(new_wheel_path)
+    new_version = Version(new_wheel.version)
+    print(f"New wheel has {new_version} and path is {new_wheel_path}")
+    return True
+
+
+def promote_targz(filename: str):
+    print(f"Found tar.gz: {filename}")
+
+    split = filename.removesuffix(".tar.gz").rsplit("/", 1)
+    base_dir = split[0]
+    dir_prefix = split[-1]
+    with tempfile.TemporaryDirectory(prefix=dir_prefix + "-") as tmp_dir:
+        print(f"  Extracting tar file to {tmp_dir}", end="")
+
+        targz = tarfile.open(filename)
+        targz.extractall(tmp_dir)
+        targz.close()
+        print(" ...done")
+
+        with open(f"{tmp_dir}/{dir_prefix}/PKG-INFO", "r") as info:
+            for line in info.readlines():
+                if "Version" in line:
+                    version = line.removeprefix("Version:").strip()
+
+        assert version, f"No version found in {filename}/PKG-INFO."
+
+        base_version = version.split("rc", 1)[0]
+
+        if any(c.isalpha() for c in base_version):
+            print(
+                f"  Base version extraction not successful and letters still in the version {base_version}."
+            )
+            print("  Only release candidates (rc) can be promoted! Aborting!")
+            return False
+        if base_version == version:
+            print(
+                f"  {version} and {base_version} are the same. Already the base version? Skipping..."
+            )
+            return False
+
+        print(
+            f"  Editing files to change version from {version} to {base_version}",
+            end="",
+        )
+
+        files_to_change = [
+            f"{tmp_dir}/{dir_prefix}/src/rocm.egg-info/requires.txt",
+            f"{tmp_dir}/{dir_prefix}/src/rocm.egg-info/PKG-INFO",
+            f"{tmp_dir}/{dir_prefix}/src/rocm_sdk/_dist_info.py",
+            f"{tmp_dir}/{dir_prefix}/PKG-INFO",
+        ]
+
+        with fileinput.input(
+            files=(files_to_change), encoding="utf-8", inplace=True
+        ) as f:
+            for line in f:
+                print(line.replace(version, base_version), end="")
+
+        print(" ...done")
+
+        print("  Creating new archive for it", end="")
+        # rename tmp dir to the new version
+        dir_prefix_no_version = dir_prefix.removesuffix(version)
+        new_dir_name = dir_prefix_no_version + base_version
+        os.rename(f"{tmp_dir}/{dir_prefix}", f"{tmp_dir}/{new_dir_name}")
+
+        print(f" {new_dir_name}", end="")
+
+        with tarfile.open(f"{base_dir}/{new_dir_name}.tar.gz", "w|gz") as tar:
+            tar.add(f"{tmp_dir}/{new_dir_name}", arcname=new_dir_name)
```

Comment:
`arcname=new_dir_name` could be improved here as this is no longer a directory and is a bit confusing in the lines after.